contracts/validator-manager/tests/PoSValidatorManagerTests.t.sol

```diff
@@ -1054,7 +1177,83 @@ abstract contract PoSValidatorManagerTest is ValidatorManagerTest {
             validationID: validationID,
             validatorOwner: address(this),
             expectedReward: expectedReward,
-            validatorWeight: DEFAULT_WEIGHT
+            validatorWeight: DEFAULT_WEIGHT,
+            rewardRecipient: address(0)
+        });
+    }
+
+    function testCompleteEndValidationWithNonValidatorRewardRecipient() public virtual {
+        bytes32 validationID = _registerDefaultValidator();
+        bytes memory setWeightMessage =
+            ValidatorMessages.packL1ValidatorWeightMessage(validationID, 1, 0);
+        bytes memory uptimeMessage = ValidatorMessages.packValidationUptimeMessage(
+            validationID, DEFAULT_COMPLETION_TIMESTAMP - DEFAULT_REGISTRATION_TIMESTAMP
+        );
+        address rewardRecipient = address(42);
+
+        _initializeEndValidation({
+            validationID: validationID,
+            completionTimestamp: DEFAULT_COMPLETION_TIMESTAMP,
+            setWeightMessage: setWeightMessage,
+            includeUptime: true,
+            uptimeMessage: uptimeMessage,
+            force: false,
+            recipientAddress: rewardRecipient
+        });
+
+        uint256 expectedReward = rewardCalculator.calculateReward({
+            stakeAmount: _weightToValue(DEFAULT_WEIGHT),
+            validatorStartTime: DEFAULT_REGISTRATION_TIMESTAMP,
+            stakingStartTime: DEFAULT_REGISTRATION_TIMESTAMP,
+            stakingEndTime: DEFAULT_COMPLETION_TIMESTAMP,
+            uptimeSeconds: DEFAULT_COMPLETION_TIMESTAMP - DEFAULT_REGISTRATION_TIMESTAMP
+        });
+
+        _completeEndValidationWithChecks({
+            validationID: validationID,
+            validatorOwner: address(this),
+            expectedReward: expectedReward,
+            validatorWeight: DEFAULT_WEIGHT,
+            rewardRecipient: rewardRecipient
+        });
+    }
+
+    function testChangeRewardRecipient() public virtual {
```

Comment:
I left comments here for a couple of different test cases, but this test is overly complicated. The `*WithNon<Validator|Delegator>RewardRecipient` tests already test that when the field is set, so I need to just check if the field is set